src/main/java/hudson/plugins/build_timeout/operations/AbortOperation.java

```diff
@@ -65,6 +65,8 @@ public boolean perform(AbstractBuild<?, ?> build, BuildListener listener, long e
         if (e != null) {
             e.interrupt(Result.ABORTED);
         }
+        build.setResult(Result.ABORTED);

```

Comment:
Yes, it doesn't work for some reason. Interruption itself works as expected, but build result is different.